src/client/session/Context.tsx

```diff
@@ -0,0 +1,137 @@
+  storeCards: CardProps[]
+  sortType: InputType
+  sortDirection: SortDirection
+  cardInput: { front: string; back: string }
+  filterChecked: boolean
+  gameState: GameState
+  buttonLabel: string
+  currentCardIndex: number
+}
+
+export type Action =
+  | { type: 'INIT_GAME'; payload: GameResultItem[] }
+  | { type: 'SET_CARD_INDEX'; payload: number }
+  | { type: 'FINISH_GAME' }
+  | { type: 'DELETE_GAME' }
+  | { type: 'SET_CARDS'; payload: GameResultItem[] }
+  | { type: 'ADD_CARD'; payload: CardProps }
+  | { type: 'DELETE_CARD'; payload: string }
+  | { type: 'SET_SORT_TYPE'; payload: InputType }
+  | { type: 'SET_SORT_DIRECTION'; payload: SortDirection }
+  | { type: 'SET_CARD_INPUT'; payload: { front: string; back: string } }
+  | { type: 'SET_FILTER_CHECKED'; payload: boolean }
+
+type ContextProps = {
+  state: State
+  dispatch: React.Dispatch<Action>
+  handleButtonClick: () => void
+}
+
+const initialState: State = {
+  gameCards: [],
+  storeCards: [],
+  gameState: GameState.NOT_STARTED,
+  buttonLabel: 'New Game',
+  currentCardIndex: 0,
+  sortType: 'front',
+  sortDirection: 'asc',
+  cardInput: { front: '', back: '' },
+  filterChecked: false,
+}
+
+const reducer = (state: State, action: Action): State => {
+  switch (action.type) {
+    case 'INIT_GAME': {
+      return {
+        ...state,
+        gameCards: action.payload,
+        gameState: GameState.ONGOING,
+        currentCardIndex: 0,
+        buttonLabel: 'Solve #1',
+      }
+    }
+    case 'SET_CARD_INDEX': {
+      let newLabel
+      if (state.gameState === GameState.ONGOING) {
+        newLabel = 'Solve #' + (action.payload + 1)
+      } else if (state.gameState === GameState.FINISHED) {
+        newLabel = 'Finished'
+      } else {
+        newLabel = 'New Game'
+      }
+
+      return {
+        ...state,
+        currentCardIndex: action.payload,
+        buttonLabel: newLabel,
+      }
+    }
+    case 'DELETE_GAME':
+      return {
+        ...state,
+        gameCards: [],
+        gameState: GameState.NOT_STARTED,
+        currentCardIndex: 0,
+        buttonLabel: 'New Game',
+      }
+    case 'FINISH_GAME':
+      return {
+        ...state,
+        gameState: GameState.FINISHED,
+        buttonLabel: 'Finished',
+      }
+    case 'SET_CARDS':
+      return { ...state, storeCards: action.payload }
+    case 'ADD_CARD':
+      return { ...state, storeCards: [...state.storeCards, action.payload] }
+    case 'DELETE_CARD':
+      return {
+        ...state,
+        storeCards: state.storeCards.filter(card => card.id !== action.payload),
+      }
+    case 'SET_SORT_TYPE':
+      return { ...state, sortType: action.payload }
+    case 'SET_SORT_DIRECTION':
+      return { ...state, sortDirection: action.payload }
+    case 'SET_CARD_INPUT':
+      return { ...state, cardInput: action.payload }
+    case 'SET_FILTER_CHECKED':
+      return { ...state, filterChecked: action.payload }
+    default:
+      return state
+  }
+}
```

Comment:
An dieser Stelle eher eine Frage. Seit meinen Änderungen in #46, wird der buttonLabel in verschiedenen cases gesetzt, und dadurch kommt 'New Game', 'Finished' und 'Solve #' mehrfach vor. 

Würde hier eine Konstante pro String sinn machen?

Reply:
Ich habs über eine kleine methode retrieveLabel gelöst, passt dir das?